app/src/main/java/com/sunasterisk/musixmatch/BaseActivity.java

```diff
@@ -0,0 +1,26 @@
+package com.sunasterisk.musixmatch;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+/**
+ * Created by superme198 on 25,March,2019
+ * in Music_Match__1.
+ */
+public abstract class BaseActivity extends AppCompatActivity {
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(getLayoutResource());
+        initComponent();
+        initData();
+
+    }
+
+    protected abstract int getLayoutResource();
```

Comment:
https://developer.android.com/reference/android/support/annotation/LayoutRes
e nên đánh dấu bằng anotation
```suggestion
    protected abstract @LayoutRes int getLayoutResource();
```